tests/test_pyrflx.py

```diff
@@ -372,6 +387,25 @@ def test_tls_length_unchecked(self) -> None:
         self.record.set("Legacy_Record_Version", "TLS_1_2")
         self.assertNotIsInstance(self.record._get_length_unchecked("Fragment"), Number)
 
+    def test_icmp_echo_request(self) -> None:
+
+        self.icmp.set("Tag", "Echo_Request")
+        self.icmp.set("Code", 0)
+        self.icmp.set("Checksum", 46957)
+        self.icmp.set("Identifier", 4)
+        self.icmp.set("Sequence_Number", 1)
+        self.icmp.set(
+            "Data",
+            b"\xe0\x28\x6d\x39\x80\x1e\x1c\x1b\x0d\xe0\xd8\xa8\x08\x00\x45\x00"
+            b"\x00\x54\x16\xf3\x40\x00\x40\x01\xe9\x73\xc0\xa8\xbc\x3d\xac\xd9"
+            b"\x10\x83\x08\x00\xb7\x6d\x00\x04\x00\x01\x6e\x5d\x67\x5e\x00\x00"
+            b"\x00\x00\xa2\xfe\x09\x00\x00\x00\x00\x00\x10\x11\x12\x13\x14\x15"
+            b"\x16\x17\x18\x19\x1a\x1b\x1c\x1d\x1e\x1f\x20\x21\x22\x23\x24\x25"
+            b"\x26\x27\x28\x29\x2a\x2b\x2c\x2d\x2e\x2f\x30\x31\x32\x33\x34\x35"
+            b"\x36\x37",
+        )
+        self.assertTrue(self.icmp.valid_message)
```

Comment:
Please test that `self.icmp.binary` equals the actual original icmp package. See `test_tlv_generate` for example.